packages/debounce/src/index.ts

```diff
@@ -13,20 +18,18 @@ import { onCleanup } from "solid-js";
  * fn('my-new-value');
  * ```
  */
-const createDebounce = <T extends (...args: unknown[]) => void, A = Parameters<T>>(
-  func: T,
+export function createDebounce<Args extends any[], F extends (...args: Args) => void>(
+  func: F,
   wait?: number
-): [fn: (...args: A extends unknown[] ? A : never) => void, clear: () => void] => {
+): F & { clear: () => void } {
```

Comment:
```suggestion
export function createDebounce<Args extends any[]>(
  func: (...args: Args) => void,
  wait?: number
): ((...args: Args) => void) & { clear: () => void }
```
-> https://github.com/davedbase/solid-primitives/pull/69#discussion_r822564838